docs/design-proposals/global-scheduler/GlobalSchedulerArchitectureDesignSpecificationFirstDraft.md

```diff
@@ -0,0 +1,416 @@
+
+In our design, all the Global Schedulers operate in a shared-state manner. That is, they shared one global resource DB which holds the cluster-level resource information of all the clusters. The DB size will be manageable since the DB only holds cluster-level resource information, such as available cluster CPU, available cluster memory, supported VM flavor list, and network topology of clusters, network proximity of clusters, etc.. Unlike the Shared-State Multi-Scheduler Architecture outlined in section 1.3.3, the global resource DB is not replicated by each Global Scheduler. Instead, they share and read/update information from one global resource DB.  In our new design, the Global Schedulers run in an optimistic concurrent way instead of using pessimistic concurrency control, which removes the locking latency.
+
+Unlike existing multi-scheduler architecture design as described in section 1.3.2 and 1.3.3, in which the incoming scheduling requests are either distributed to all schedulers or a scheduler in a round-robin way, in our design, the incoming scheduling requests are distributed to a Global Scheduler based on the request's profile. Section 2.5.4 illustrates how this distribution mechanism combined with the Global Scheduler partitioning scheme (Section 2.5.3) will mitigate the scheduling conflict when all the Global Schedulers run concurrently.
+
+In summary, this design eliminates the issues associated with existing Two-Level Scheduler Architecture, Siloed Multi-Scheduler Architecture, and Shared-State Multi-Scheduler Architecture. It renders good scalability, low latency, and global optimal scheduling while satisfying the user-supplied placement constraints.
+
+On the southbound, it integrates with different types of clusters via a flexible plugin design.
+
+Figure.1 is an illustration of the high-level architecture.
+![image.png](/images/2.3.png)
+
+In the future, we may explore whether this two-level architecture may flatten into a one-level architecture.
+
+## 2.4 Architecture Component Description
+
+The following table describes each component/subsystem in the Global Scheduling Architecture.
+
+<table>
+  <tr>
+    <th>Subsystem or Component</th>
+    <th>Description</th>
+    <th>Roles and Responsibilities</th>
+  </tr>
+  
+  <tr>
+    <td>API Server</td>
+    <td>
+    	API server exposes VM/Container CRUD APIs to clients, who use the Global Scheduler Service to request/read/update/delete VM/Container resource
```

Comment:
The API of global scheduler may only support schedule ability like allocation /release.Scheduler itself cannot do the CRUD of VM.

Reply:
Ok. will update